<launch>

  <!-- Arguments -->
  <arg name="bag_file" value="/home/bjoshi/Downloads/pamir_shipwreck_sync.bag"/>
  <arg name="image_dir" value="/home/bjoshi/colmap/pamir_shipwreck/images"/>
  <arg name="traj_file" value="/home/bjoshi/pamir_shipwreck.txt"/>
  <arg name="stereo" value="true"/>
  <arg name="left_image_topic" value="/slave1/image_raw"/>
  <arg name="right_image_topic" value="/slave2/image_raw"/>
  <arg name="config_file" value="$(find utils_ros)/config/stereo_rig/shipwreck.yaml"/>

  <arg name="scale" value="1.0"/>
  <arg name="compressed" value="true"/>

  <node name="extract_keyframes" pkg="utils_ros" type="write_kf_images" output="screen">

    <param name="bag_file" value="$(arg bag_file)"/>
    <param name="image_dir" value="$(arg image_dir)"/>
    <param name="traj_file" value="$(arg traj_file)"/>
    <param name="stereo" value="$(arg stereo)"/>
    <param name="left_image_topic" value="$(arg left_image_topic)"/>
    <param name="right_image_topic" value="$(arg right_image_topic)"/>
    <param name="config_file" value="$(arg config_file)"/>

    <param name="scale" value="$(arg scale)"/>
    <param name="compressed" value="$(arg compressed)"/>

  </node>

</launch>

